### PR TITLE
python: Improve yapf formatting

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: python: Improve yapf formatting
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
this adds a couple of rules to the yapf formatting that make it easier to stick within line length limits and improve readability of code with hanging indents

File: tools/type_whisperer/typedb_gen.py [hunk 1]
```diff
@@ -20,8 +20,9 @@
     ('envoy\.type\.matcher', 'envoy.type.matcher.v3'),
     ('envoy\.type', 'envoy.type.v3'),
     ('envoy\.config\.cluster\.redis', 'envoy.extensions.clusters.redis.v3'),
-    ('envoy\.config\.retry\.previous_priorities',
-     'envoy.extensions.retry.priority.previous_priorities.v3'),
+    (
+        'envoy\.config\.retry\.previous_priorities',
+        'envoy.extensions.retry.priority.previous_priorities.v3'),
```
Comment (phlax) [inline tools/type_whisperer/typedb_gen.py:25]: this is kinda annoying

File: .azure-pipelines/pipelines.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ stages:
             displayName: "Run format pre-checks"
 
       - job: format
-        dependsOn: ["format_pre"]
+        dependsOn: [] # "format_pre"]
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:31]: remove

File: tools/type_whisperer/typedb_gen.py (hunk 1 above)
Comment (phlax) [reply tools/type_whisperer/typedb_gen.py:25]: after playing with the "knobs" i cant seem to find any that prevent this (other than disabling `split_before_first_argument` altogether)

i have raised a ticket here https://github.com/google/yapf/issues/907

**Review by htuch — APPROVED** [review]
@phlax this is great. When do you think we will have stable Python formatting? I'm a bit worried about in-flight PRs (e.g. #15564). Is there a way to avoid too much more churn?

**Pull request merged**